src/main/java/com/coveo/saml/SamlClient.java

```diff
@@ -281,14 +318,38 @@ public SamlResponse processPostFromIdentityProvider(HttpServletRequest request)
   public static SamlClient fromMetadata(
       String relyingPartyIdentifier, String assertionConsumerServiceUrl, Reader metadata)
       throws SamlException {
+    return fromMetadata(relyingPartyIdentifier, assertionConsumerServiceUrl, metadata, false);
+  }
+
+  /**
+   * Constructs an SAML client using XML metadata obtained from the identity provider. <p> When
+   * using Okta as an identity provider, it is possible to pass null to relyingPartyIdentifier and
+   * assertionConsumerServiceUrl; they will be inferred from the metadata provider XML.
+   *
+   * @param relyingPartyIdentifier      the identifier for the relying party.
+   * @param assertionConsumerServiceUrl the url where the identity provider will post back the
+   *                                    SAML response.
+   * @param metadata                    the XML metadata obtained from the identity provider.
+   * @param redirect                    If HTTP-Redirect should be used instead of HTTP-Post
+   * @return The created {@link SamlClient}.
+   * @throws SamlException thrown if any error occur while loading the metadata information.
+   */
+  public static SamlClient fromMetadata(
+      String relyingPartyIdentifier, String assertionConsumerServiceUrl, Reader metadata,
+      Boolean redirect)
+      throws SamlException {
 
     ensureOpenSamlIsInitialized();
 
     MetadataProvider metadataProvider = createMetadataProvider(metadata);
     EntityDescriptor entityDescriptor = getEntityDescriptor(metadataProvider);
 
     IDPSSODescriptor idpSsoDescriptor = getIDPSSODescriptor(entityDescriptor);
-    SingleSignOnService postBinding = getPostBinding(idpSsoDescriptor);
+    SingleSignOnService idpBinding = null;
+    if (redirect)
```

Comment:
Not sure this is appropriate... Shouldn't the code just figure out the binding from the metadata? Like, pick the first one it finds? Or does your metadata contain a POST binding but you want to force usage of the GET one?

Reply:
The IdP Metadata that I'm looking at, from the CAS Server implementation of a SAML2 IdP, contains both the HTTP-POST and the HTTP-Redirect information in it, and it is (seems to be) up to the client to choose which one to use.  I tried using that IdP Metadata and just changing the URL to the HTTP-Redirect and appending the ?SAMLRequest=<encoded metadata> to that, and I get an error (Error Unmarshalling Metadata, or something like that).  If what I'm reading on the Internet is correct, it's because you can't just change the URL, you have to tell the client to use the HTTP-Redirect method, and it should then choose the correct URL.

I'm fairly new to SAML, though, so if you think this should be otherwise, let me know.

Reply:
OK. I'm no SAML specialist either in fact... Then it makes sense to do as you did initially. Nevermind my comment.